pkg/controller/cassandradatacenter/controller/jvm.go

```diff
@@ -0,0 +1,113 @@
+package controller
+
+import (
+	cassandraoperatorv1alpha1 "github.com/instaclustr/cassandra-operator/pkg/apis/cassandraoperator/v1alpha1"
+	"strconv"
+	"strings"
+)
+
+func cassandraJVMOptions(cdc *cassandraoperatorv1alpha1.CassandraDataCenter) (*string, error) {
+	// TODO: should this be Limits or Requests?
+	jvmHeapSize := jvmHeapSize(memoryLimit(cdc))
+
+	var writer strings.Builder
+
+	if err := writeFormattedProperties(&writer, map[string]string{
+		"-Xms%s\n": strconv.FormatInt(jvmHeapSize, 10),
+		"-Xmx%s\n": strconv.FormatInt(jvmHeapSize, 10),
+	}); err != nil {
+		return nil, err
+	}
+
+	// copied from stock jvm.options
+	if !isUsingGC1(jvmHeapSize) {
+
+		// young gen size
+		if err := writeFormattedProperty(&writer, "-Xmn%s\n", strconv.FormatInt(youngGen(jvmHeapSize), 10)); err != nil {
```

Comment:
There is also inconsistency here now too. Instead of everything just using `fprintf`/`fprintln`, we now have 3 functions: `writeFormattedProperty`, `writePropertiesWithNewLine`, and `writePropertiesConditionally`.

Why doesn't `writeFormattedProperty` add a new line and instead still uses `\n` rather than the `NewLine` constant? Does `writePropertiesConditionally` add new lines? Why doesn't everything just use `writePropertiesConditionally`, since the condition is apparently optional?

This is far, far, far more complex than the original, which was simply formatting some stings into a buffer, with a few `if`s thrown in if things were conditional!